tests/common/include/tests/common/tools_propagator.inl

```diff
@@ -156,61 +113,71 @@ TEST_P(PropagatorWithRkStepper, propagator_rk_stepper) {
     auto d = create_toy_geometry(host_mr, n_brl_layers, n_edc_layers);
 
     // Create the navigator
-    using detray_navigator = navigator<decltype(d)>;
-    using detray_b_field = constant_magnetic_field<>;
-    using detray_stepper = rk_stepper<detray_b_field, free_track_parameters>;
-    using detray_propagator = propagator<detray_stepper, detray_navigator>;
+    using navigator_t = navigator<decltype(d), navigation::print_inspector>;
+    using b_field_t = constant_magnetic_field<>;
+    using stepper_t = rk_stepper<b_field_t, free_track_parameters>;
+    using propagator_t = propagator<stepper_t, navigator_t>;
 
     // Constant magnetic field
-    // vector3 B{0, 0, 2 * unit_constants::T};
     vector3 B = GetParam();
-    detray_b_field b_field(B);
+    b_field_t b_field(B);
 
-    // Set initial position and momentum of tracks
-    const point3 ori{0., 0., 0.};
+    stepper_t s(b_field);
+    navigator_t n(d);
+    propagator_t p(std::move(s), std::move(n));
 
-    detray_stepper s(b_field);
-    detray_navigator n(std::move(d));
-    detray_propagator p(std::move(s), std::move(n));
+    // Set origin position of tracks
+    const point3 ori{0., 0., 0.};
 
     // Loops of theta values ]0,pi[
     for (unsigned int itheta = 0; itheta < theta_steps; ++itheta) {
         scalar theta = 0.001 + itheta * (M_PI - 0.001) / theta_steps;
         scalar sin_theta = std::sin(theta);
         scalar cos_theta = std::cos(theta);
-
         // Loops of phi values [-pi, pi]
         for (unsigned int iphi = 0; iphi < phi_steps; ++iphi) {
             // The direction
             scalar phi = -M_PI + iphi * (2 * M_PI) / phi_steps;
             scalar sin_phi = std::sin(phi);
             scalar cos_phi = std::cos(phi);
-            vector3 mom{cos_phi * sin_theta, sin_phi * sin_theta, cos_theta};
-            mom = 10. * mom;
 
             // intialize a track
+            vector3 mom{cos_phi * sin_theta, sin_phi * sin_theta, cos_theta};
+            vector::normalize(mom);
+            mom = 10. * unit_constants::GeV * mom;
             free_track_parameters traj(ori, 0, mom, -1);
-            helix_gun helix(traj, B);
-            combined_inspector ci{helix_inspector(helix), print_inspector{}};
+            traj.set_overstep_tolerance(-10 * unit_constants::um);
 
-            detray_propagator::state state(traj);
+            helix_gun helix(traj, B);
+            combined_inspector ci{helix_inspector(helix),
+                                  propagation::print_inspector{}};
 
-            p.propagate(state, ci);
+            propagator_t::state state(traj);
+            state._stepping.set_path_limit(path_limit);
 
-            // Ensure that the tracks reach the end of the world
-            EXPECT_EQ(state._navigation.volume(), dindex_invalid);
```

Comment:
Why did you kill this line?

Reply:
Because the line below is a more thorough check of whether the propagation exited correctly. It uses a new member function of the navigator in the propagator ```return n_state.is_complete();```